chef-config/lib/chef-config/path_helper.rb

```diff
@@ -55,18 +55,24 @@ def self.path_separator
       end
     end
 
-    def self.join(*args)
-      path_separator_regex = Regexp.escape(File::SEPARATOR)
-      unless path_separator == File::SEPARATOR
-        path_separator_regex << Regexp.escape(path_separator)
-      end
+    def self.path_separator_regex
+      @path_separator_regex ||=
+        begin
+          path_separator_regex = Regexp.escape(File::SEPARATOR)
+          unless path_separator == File::SEPARATOR
+            path_separator_regex << Regexp.escape(path_separator)
+          end
+          path_separator_regex
+        end
+    end
```

Comment:
Since this wasn't a function already, it could be a local variable rather than exposing a new thing externally:

```suggestion
    path_seperator_regex = [Regexp.escape(File::SEPARATOR), Regexp.escape(path_separator)].uniq.join
```